kedro/contrib/io/pyspark/spark_hive_data_set.py

```diff
@@ -0,0 +1,276 @@
+                    table=self._table,
+                )
+            )
+
+    @staticmethod
+    def _get_spark():
+        return SparkSession.builder.getOrCreate()
+
+    def _load(self) -> DataFrame:
+        return self._get_spark().sql(
+            "select * from {database}.{table}".format(
+                database=self._database, table=self._table
+            )
+        )
+
+    def _save(self, data: DataFrame) -> None:
+        self._validate_save(data)
+        write_methods = {
+            "insert": self._insert_save,
+            "upsert": self._upsert_save,
+            "overwrite": self._overwrite_save,
+        }
+        write_methods[self._write_mode](data)
+
+    def _insert_save(self, data):
+        data.createOrReplaceTempView("tmp")
+        self._get_spark().sql(
+            "insert into {database}.{table} select {columns} from tmp".format(
+                database=self._database,
+                table=self._table,
+                columns=", ".join(self._table_columns),
+            )
+        )
+
+    def _upsert_save(self, data: DataFrame):
+        if self._load().rdd.isEmpty():
+            self._insert_save(data)
+        else:
+            joined_data = data.alias("new").join(
+                self._load().alias("old"), self._table_pk, "outer"
+            )
+            upsert_dataset = joined_data.select(
+                [
+                    coalesce(
+                        "new.{}".format(col_name), "old.{}".format(col_name)
+                    ).alias(col_name)
+                    for col_name in data.columns
+                    if col_name not in self._table_pk
+                ]
+                + self._table_pk
+            )
+            temporary_persisted_tbl = "temp_{}".format(uuid.uuid4().int)
+            with StagedHiveDataSet(
+                upsert_dataset,
+                stage_database_name=self._database,
+                stage_table_name=temporary_persisted_tbl,
+            ) as temp_table:
+                self._overwrite_save(temp_table.staged_data)
+
+    def _overwrite_save(self, data):
+        self._get_spark().sql(
+            "truncate table {database}.{table}".format(
+                database=self._database, table=self._table
+            )
+        )
+        self._insert_save(data)
+
+    def _validate_save(self, data):
+        hive_dtypes = self._load().dtypes
+        if data.dtypes != hive_dtypes:
+            new_cols = [i for i in data.dtypes if i not in hive_dtypes]
+            missing_cols = [i for i in hive_dtypes if i not in data.dtypes]
+            raise DataSetError(
+                "dataset does not match hive table schema.\n"
+                "Present on insert only: {new_cols}\n"
+                "Present on schema only: {missing_cols}".format(
+                    new_cols=new_cols, missing_cols=missing_cols
+                )
+            )
+
+    def _exists(self) -> bool:
+        if self._database in [
```

Comment:
List databases gives error because it tries to get access to all the databases belongs to anyone I think but show databases only looks for the databases accessed by the current user from hive metastore.
```
        if self._get_spark().sql("show databases").filter(col("databaseName") == self._database).count() == 1:
```

Reply:
Good catch, thanks for this - has been updated with the suggested fix.